gtsfm/utils/geometry_comparisons.py

```diff
@@ -30,40 +30,40 @@ def align_rotations(input_list: List[Rot3], ref_list: List[Rot3]) -> List[Rot3]:
     return [w2Rw1.compose(w1Ri) for w1Ri in input_list]
 
 
-def align_poses(input_list: List[Pose3], ref_list: List[Pose3]) -> List[Pose3]:
-    """Aligns the list of poses to the reference list by shifting origin and
-    scaling translations.
+def align_poses_sim3(aTi_list: List[Pose3], bTi_list: List[Pose3]) -> List[Pose3]:
+    """Align by similarity transformation.
 
-    Args:
-        input_list: input poses which need to be aligned, suppose w1Ti in world-1 frame for all frames i.
-        ref_list: reference poses which are target for alignment, suppose w2Ti_ in world-2 frame for all frames i.
+    We force SIM(3) alignment rather than SE(3) alignment.
+    We assume the two trajectories are of the exact same length.
 
+    Args:
+        aTi_list: reference poses in frame "a" which are the targets for alignment
+        bTi_list: input poses which need to be aligned to frame "a"
+            
     Returns:
-        transformed poses which have the same origin and scale as reference (now living in world-2 frame)
+        aTi_list_: transformed input poses previously "bTi_list" but now which
+            have the same origin and scale as reference (now living in "a" frame)
     """
-    # match the scales first
-    wTi0 = input_list[0]
-    input_distances = np.array([np.linalg.norm((wTi.between(wTi0)).translation()) for wTi in input_list[1:]])
+    n_to_align = len(aTi_list)
+    assert len(aTi_list) == len(bTi_list)
+    assert n_to_align >= 2, "SIM(3) alignment uses at least 2 frames"
```

Comment:
Do you think raising `ValueError` is more appropriate than raising `AssertionError`

Reply:
Not too sure to be honest